core/src/plugin/mod.rs

```diff
@@ -20,25 +20,28 @@ pub trait Plugin: Sized + Send + Sync + 'static {
     /// The type of the port container.
     type Ports: PortContainer;
 
+    /// The host features used by this plugin.
+    type Features: FeatureCollection<'static>;
+
     /// Create a new plugin instance.
     ///
     /// This method only creates an instance of the plugin, it does not reset or set up it's internal state. This is done by the `activate` method.
-    fn new(plugin_info: &PluginInfo, features: FeatureContainer) -> Self;
+    fn new(plugin_info: &PluginInfo, features: &Self::Features) -> Self;
```

Comment:
Why do you pass a reference to the features struct, not it's ownership? It is not used by `PluginInstance` and the plugin can decide if it wants to store it or not. Having to pass a reference every time also makes the argument lists of the other methods longer.

Reply:
I initially thought it could be possible to have a case where references to the plugin are valid but not the ones to the features, however thinking about it I don't really see it. I can only see the opposite (workers), and whenever that happens I believe we can keep a copy of the resolved features for them to use if needed.

I'll change it back, then!